week 2/script.js

```diff
@@ -0,0 +1,219 @@
+let currencyRates = {
+    timestamp: 1519296206,
+    base: "EUR",
+    date: "2021-03-17",
+    rates: {
+        USD: 1.23396
+    }
+};
+
+class CurrencyRate {
+    constructor(code, rate, date) {
+        this.code = code;
+        this.rate = rate;
+        this.date = date;
+    }
+}
+
+const currencyRatesArray = Object.entries(currencyRates.rates).map(([code, rate, date]) => new CurrencyRate(code, rate, currencyRates.date));
```

Comment:
You are making your life complicated here. The array of currency rates could simply be
```js
const currencyRatesArray = [
  {
    timestamp: 1519296206,
    base: "EUR",
    date: "2021-03-17",
    rates: {
        USD: 1.23396
    }
  },
  {
    timestamp: 1519296206,
    base: "DKK",
    date: "2021-03-17",
    rates: {
        SEK: 0.987
    }
  }
];
```